pkg/fanal/analyzer/analyzer.go

```diff
@@ -454,19 +448,24 @@ func (ag AnalyzerGroup) AnalyzeFile(ctx context.Context, wg *sync.WaitGroup, lim
 	return nil
 }
 
+// RequiredPostAnalyzers returns a list of analyzer types that require the given file.
 func (ag AnalyzerGroup) RequiredPostAnalyzers(filePath string, info os.FileInfo) []Type {
 	if info.IsDir() {
 		return nil
 	}
 	var postAnalyzerTypes []Type
 	for _, a := range ag.postAnalyzers {
-		if a.Required(filePath, info) {
+		if ag.filePatternMatch(a.Type(), filePath) || a.Required(filePath, info) {
 			postAnalyzerTypes = append(postAnalyzerTypes, a.Type())
 		}
 	}
 	return postAnalyzerTypes
 }
 
+// PostAnalyze passes a virtual filesystem containing only required files
+// and passes it to the respective post-analyzer.
+// The obtained results are merged into the "result".
+// This function may be called concurrently and must be thread-safe.
 func (ag AnalyzerGroup) PostAnalyze(ctx context.Context, files *syncx.Map[Type, *mapfs.FS], result *AnalysisResult, opts AnalysisOptions) error {
 	for _, a := range ag.postAnalyzers {
 		fsys, ok := files.Load(a.Type())
```

Comment:
Maybe we can add a logger print here as this !ok will be true in case there are no relevant files found.

```go
log.Logger.Debugf("analyzer %s has no relevant files, skipping", a.Type())
```

Reply:
A lot of analyzers are present in Trivy, and the log messages will be overwhelming as most analyzers don't find relevant files in a single scan. Instead, I've added a debug message showing which misconfiguration scanning is performed.
https://github.com/aquasecurity/trivy/pull/4090/commits/4ebf381e7b081f95862d1927b71fe254101224ec